src/smt/smt_engine.cpp

```diff
@@ -1208,11 +1212,45 @@ void SmtEngine::setDefaults() {
   {
     d_logic = LogicInfo("QF_NIA");
   }
-  else if ((d_logic.getLogicString() == "QF_UFBV"
-            || d_logic.getLogicString() == "QF_ABV")
-           && options::bitblastMode() == theory::bv::BITBLAST_MODE_EAGER)
+
+  // set options about ackermannization
+  if (options::ackermann())
```

Comment:
This condition would actually move inside the check for `options::produceModels()` (see the code [my comment](https://github.com/CVC4/CVC4/pull/3315/files#r331704039)). Let's discuss briefly tomorrow to make sure that we are on the same page :)